build/metadata_utilities.py

```diff
@@ -37,46 +121,108 @@ def test_merge_dict_second_is_empty():
     a        = {'a':1, 'b':2}
     b        = {}
     expected = {'a':1, 'b':2}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_simple():
     a        = {'a':1, 'b':2}
     b        = {'c':3}
     expected = {'a':1, 'b':2, 'c':3}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_first_is_empty():
     a        = {}
     b        = {'a':1, 'b':2}
     expected = {'a':1, 'b':2}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_key_exists():
     a        = {'a':1, 'b':2}
     b        = {'b':3, 'c':4}
     expected = {'a':1, 'b':3, 'c':4}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_recurse():
     a        = {'a':1, 'b':{'b1':5, 'b2':6}}
     b        = {'b':{'b3':7}, 'c':4}
     expected = {'a':1, 'b':{'b1':5, 'b2':6, 'b3':7}, 'c':4}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_replace_in_list():
     a        = {'a':1, 'b':{'b1':5, 'b2':6}, 'c':['x', 'y', 'z']}
     b        = {'b':{'b3':7, 'b1':8}, 'c':{0:'r', 1:'s', 2:'t'}}
     expected = {'a':1, 'b':{'b1':8, 'b2':6, 'b3':7}, 'c':['r', 's', 't']}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_replace_in_dict_and_list():
     a        = {'a':1, 'b':{'b1':2, 'b2':3}, 'c':['x', 'y', 'z']}
     b        = {'b':{'b3':7, 'b1':8}, 'c':{0:'r', 1:'s', 2:'t'}}
     expected = {'a':1, 'b':{'b1':8, 'b2':3, 'b3':7}, 'c':['r', 's', 't']}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
 
 def test_merge_dict_with_regex():
     a        = {'aaa':{}, 'aaaa':{'x':98}, 'aaaaa':{}, 'bbb':{'bbb1':2, 'bbb2':3}, 'ccc':['x', 'y', 'z']}
     b        = {'a+':{'z':99}}
     expected = {'aaa':{'z':99}, 'aaaa':{'x':98, 'z':99}, 'aaaaa':{'z':99}, 'bbb':{'bbb1':2, 'bbb2':3}, 'ccc':['x', 'y', 'z']}
-    _do_the_test(a,b, expected)
+    _do_the_test_merge_dicts(a,b, expected)
+
+def _do_the_test_add_all_metadata(funtions, expected):
+    actual = copy.deepcopy(funtions)
```

Comment:
I didn't do deep copy in line 116. Did I screw up?

Reply:
I would say, if it worked, you didn't do it wrong. When I tried using a copy(), it didn't work,

In your function, is `a != actual`? When I didn't use `deepcopy()`, actual and function would end up being the same